src/main/java/com/emc/pravega/perf/PravegaPerfTest.java

```diff
@@ -259,131 +314,278 @@ private static void parseCmdLine(String[] args) {
                     runKafka = Boolean.parseBoolean(commandline.getOptionValue("kafka"));
                 }
 
+                if (commandline.hasOption("segments")) {
+                    segmentCount = Integer.parseInt(commandline.getOptionValue("segments"));
+                } else {
+                    segmentCount = producerCount; 
+                }
+
+                if (commandline.hasOption("recreate")) {
+                    recreate = Boolean.parseBoolean(commandline.getOptionValue("recreate"));
+                }
+
             }
         } catch (Exception nfe) {
             System.out.println("Invalid arguments. Starting with default values");
             nfe.printStackTrace();
         }
     }
 
-    /**
-     * A Sensor simulator class that generates dummy value as temperature measurement and ingests to specified stream.
-     */
 
-    private static class WriterWorker implements Runnable {
+    private static class StartLocalService {
+        static final int PORT = 9090;
+        static final String SCOPE = "Scope";
+        static final String STREAM_NAME = "aaj";
+    }
+}
+
+class StreamHandler{
+      String scope;
+      String stream;
+      String controllerUri;   
+      ControllerImpl controller; 
+      StreamManager streamManager;
+      StreamConfiguration streamconfig;
+      ReaderGroupManager readerGroupManager; 
+      ReaderGroup readerGroup;
+      ScheduledExecutorService bgexecutor;
+      int segCount;
+      int timeout;
+
+
+      StreamHandler(String scope, String stream, 
+                    String uri, int segs,
+                    int timeout,  ControllerImpl contrl,
+                    ScheduledExecutorService bgexecutor) throws Exception {
+               this.scope = scope;
+               this.stream = stream;
+               this.controllerUri= uri;
+               this.controller = contrl;
+               this.segCount = segs; 
+               this.timeout = timeout;
+               this.readerGroup = null;
+               this.bgexecutor = bgexecutor;
+               streamManager = StreamManager.create(new URI(uri));
+               streamManager.createScope(scope);
+               streamconfig = null;
+
+       } 
+
+
+       boolean create() throws Exception {
```

Comment:
Only declare the exception types your code actually throws. Here and below.

Reply:
yes , done